### DISCUSSION TITLE
I miss having a sense of homoiconicity in pkl

### DISCUSSION THREAD

**Discussion by fhalde** [apple/pkl#648] [Ideas]
pkl must be able to output the configuration in a way that can be slurp'd ( read ) back into pkl types directly without much of a fuss.

```
class World {
 wname: String
}

class Hello {
 hname: String
 world: World
}

hello: Hello = new Hello {
 hname = "hello"
 world = new World {
  wname = "world"
 }
}
```

```
pkl eval foo.pkl > foo.pcf
hello {
  hname = "hello"
  world {
    wname = "world"
  }
}
```

How can I re-read/import foo.pcf in a different pkl file that directly instantiates into a Hello type. The closest I got to was [deepToTyped](https://github.com/apple/pkl-pantry/blob/main/packages/pkl.experimental.deepToTyped/deepToTyped.pkl)

Ideally, we were looking for a way to build toolings to work with the generated code to create pkl objects, serialize them which can then be read by other pkl configurations.

Is this already possible? deepToTyped works fine for our use-case though

The other only viable alternative is to store reusable pkl objects as modules
```
hello: Hello = new Hello {
 hname = "hello"
 world = new World {
  wname = "world"
 }
}
```
unfortunately this style is not programmable?

**Comment by bioball** [comment]
We are considering providing a `PklRenderer`, which can do what you're looking for here. Until then, something like `deepToTyped` is probably the best option.
